dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/kdoc/ResolveKDocLink.kt

```diff
@@ -78,5 +78,20 @@ internal fun KtAnalysisSession.resolveKDocLinkToDRI(link: KDocLink): DRI? {
 
 private fun KtAnalysisSession.resolveToSymbol(kDocLink: KDocLink): KtSymbol? {
     val lastNameSegment = kDocLink.children.filterIsInstance<KDocName>().lastOrNull()
-    return lastNameSegment?.mainReference?.resolveToSymbols()?.firstOrNull()
+    return lastNameSegment?.mainReference?.resolveToSymbols()?.sortedWith(linkCandidatesComparator)?.firstOrNull()
 }
+
+/**
+ * The order is like in [K1](https://github.com/JetBrains/intellij-community/blob/84f54ed97da66d6e24e6572345869bf1071945b6/plugins/kotlin/base/fe10/kdoc/src/org/jetbrains/kotlin/idea/kdoc/resolveKDocLink.kt#L104)
+ *
+ * TODO KT-64190
+ */
+private var linkCandidatesComparator: Comparator<KtSymbol> = compareBy{
+    when(it) {
+        is KtClassifierSymbol -> 1
+        is KtPackageSymbol -> 2
+        is KtFunctionSymbol -> 3
+        is KtVariableSymbol-> 4
+        else -> 5
+    }
+}
```

Comment:
Could you please cover all cases with tests? I see a test for the initial bug where class < function, which is good, and it would be very helpful to make sure other cases don't break either. Especially once KT-64190 is fixed - it will help us verify that it was implemented the way we expected

And a side question: can there be two symbols of the same type? For example, two `KtClassifierSymbol`? Asking in case there's a non-obvious corner case related to source sets, expect/actual, `internal` visibility, java interop or something else? If there is, does it make sense to add a second comparison with `thenBy { it.name }` or something similar, to make it predictable?

Reply:
> can there be two symbols of the same type? 

It is possible for `KtFunctionSymbol` when there are overloads. In this case, the order is not specified at all and it does not affect a result URL. It is not easy to make the order like in K1 on Dokka's side and consistent with IDEA. See added `tests for overloads`

For the other types of symbols, it is difficult to come up with a case when we have the same name in the same scope.

Reply:
I think a stable order of overloads can be addressed in #3250